ui/app/models/totp.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Model, { attr } from '@ember-data/model';
+import { alias } from '@ember/object/computed';
+import { computed } from '@ember/object';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import fieldToAttrs, { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+
+const ALGORITHMS = ['SHA1', 'SHA256', 'SHA512'];
+const DIGITS = [6, 8];
+const SKEW = [0, 1];
+
+const validations = {
+  account_name: [
+    { type: 'presence', message: "Account name can't be blank." },
+    {
+      type: 'containsWhiteSpace',
+      message:
+        "Account name contains whitespace. If this is desired, you'll need to encode it with %20 in API requests.",
+      level: 'warn',
+    },
+  ],
+  algorithm: [
+    {
+      validator: (value) => ALGORITHMS.includes(value),
+      message: 'Algorithm must be one of ' + ALGORITHMS.join(', '),
+    },
+  ],
+  digits: [
+    { validator: (value) => DIGITS.includes(value), message: 'Digits must be one of ' + DIGITS.join(', ') },
+  ],
+  period: [{ type: 'number', message: 'Period must be a number.' }],
+
+  skew: [{ validator: (value) => SKEW.includes(value), message: 'Skew must be one of ' + SKEW.join(', ') }],
+};
+
+@withModelValidations(validations)
+export default class TotpModel extends Model {
+  @attr('string', {
+    readOnly: true,
+  })
+  backend;
+
+  @alias('account_name') name;
+
+  @attr('string', {
+    fieldValue: 'name',
+    editDisabled: true,
+  })
+  account_name;
+  @attr('string', {
+    editDisabled: true,
+    possibleValues: ALGORITHMS,
+    defaultValue: 'SHA1',
```

Comment:
Is it possible to provide default values for the create/edit form without having to set them here on the model?

Reply:
Sort of -- do you want the default value to be dynamic, or why isn't setting it on the model preferable?

Reply:
No, the main reason was that everything which is set on the model is currently serialized. But some options are mutually exclusive in the `POST` request. I just realized that it's probably better to tackle this in the serializer and conditionally remove attributes from the serialized data.
The other reason was that some attributes are only used in the `POST` request but never returned by the `GET` request. But with the default values it looks like they got returned from the request as well.

Reply:
Got it! Yes, we don't want to show "default" values that are not actually reflected in the backend. One way we get around this limitation (which we run into fairly often) is to instead set the "initial values" in the `createRecord` invocation. So in this case, in the credentials route, we would have: 
```
getTOTPCode(backend, keyName) {
    return this.store.adapterFor('totp').generateCode(backend, keyName, { algorithm: 'SHA1' });
  },
```
with whatever other values we want to set initially, and remove the `defaultValue` unless this is truly the default value that the backend has. Good question!